drivers/accel/adxl367/adxl367.c

```diff
@@ -0,0 +1,1310 @@
+					 xyz_values,
+					 ADXL367_REG_XDATA_H,
+					 6);
+
+	// result is 14 bits long, ignore last 2 bits from low byte
+	*x = ((int16_t)xyz_values[0] << 6) + (xyz_values[1] >> 2);
+	*y = ((int16_t)xyz_values[2] << 6) + (xyz_values[3] >> 2);
+	*z = ((int16_t)xyz_values[4] << 6) + (xyz_values[5] >> 2);
+	// extend sign to 16 bits
+	if (*x & NO_OS_BIT(13))
+		*x |= NO_OS_GENMASK(15, 14);
+	if (*y & NO_OS_BIT(13))
+		*y |= NO_OS_GENMASK(15, 14);
+	if (*z & NO_OS_BIT(13))
+		*z |= NO_OS_GENMASK(15, 14);
+
+	return 0;
+}
+
+/***************************************************************************//**
+ * @brief Converts raw acceleration value to g value.
+ *
+ * @param dev       - The device structure.
+ * @param raw_accel - Raw acceleration value.
+ * @param g_val		- Struct to store converted data.
+ *
+ * @return 0 in case of success, negative error code otherwise.
+*******************************************************************************/
+static int32_t adxl367_accel_conv(struct adxl367_dev *dev,
+				  uint16_t raw_accel, struct adxl367_fractional_val* g_val)
+{
+	int64_t accel_data;
+
+	accel_data = (int16_t)raw_accel;
+
+	// Apply scale factor based on the selected range
+	accel_data = accel_data * ADXL367_ACC_SCALE_FACTOR_MUL *
+		     adxl367_scale_mul[dev->range];
+
+	g_val->integer = no_os_div_s64_rem(accel_data,
+					   ADXL367_ACC_SCALE_FACTOR_DIV, &(g_val->fractional));
+
+	return 0;
+}
+
+/***************************************************************************//**
+ * @brief Reads the 3-axis raw data from the accelerometer and converts it to g.
+ *
+ * @param dev - The device structure.
+ * @param x   - Stores the X-axis data.
+ * @param y   - Stores the Y-axis data.
+ * @param z   - Stores the Z-axis data.
+ *
+ * @return 0 in case of success, negative error code otherwise.
+*******************************************************************************/
+int32_t adxl367_get_g_xyz(struct adxl367_dev *dev,
+			  struct adxl367_fractional_val* x,
+			  struct adxl367_fractional_val* y,
+			  struct adxl367_fractional_val* z)
+{
+	int32_t ret;
+	int16_t x_raw, y_raw, z_raw;
+
+	ret = adxl367_get_raw_xyz(dev, &x_raw, &y_raw, &z_raw);
+	if (ret)
+		return ret;
+
+	ret = adxl367_accel_conv(dev, x_raw, x);
+	if (ret)
+		return ret;
+	ret = adxl367_accel_conv(dev, y_raw, y);
+	if (ret)
+		return ret;
+	return adxl367_accel_conv(dev, z_raw, z);
+}
+
+/***************************************************************************//**
+ * @brief Enables temperature reading.
+ *
+ * @param dev    - The device structure.
+ * @param enable - 1 - ENABLE
+ * 				   2 - DISABLE
+ *
+ * @return 0 in case of success, negative error code otherwise.
+*******************************************************************************/
+int32_t adxl367_temp_read_en(struct adxl367_dev *dev, uint8_t enable)
+{
+	return adxl367_reg_write_msk(dev,
+				     ADXL367_REG_TEMP_CTL,
+				     enable ? ADXL367_TEMP_EN : 0U,
```

Comment:
I was wondering whether having a conditional statement within a function call is appropriate. Doesn't it affect code readability?